src/cephadm/cephadm.py

```diff
@@ -3359,23 +3359,26 @@ def extract_uid_gid(ctx, img='', file_path='/var/lib/ceph'):
     raise RuntimeError('uid/gid not found')
 
 
-def deploy_daemon(ctx, fsid, daemon_type, daemon_id, c, uid, gid,
-                  config=None, keyring=None,
-                  osd_fsid=None,
-                  reconfig=False,
-                  ports=None):
-    # type: (CephadmContext, str, str, Union[int, str], Optional[CephContainer], int, int, Optional[str], Optional[str], Optional[str], Optional[bool], Optional[List[int]]) -> None
+def deploy_daemon(ctx: CephadmContext, fsid: str, daemon_type: str,
+                  daemon_id: Union[int, str], c: Optional['CephContainer'],
+                  uid: int, gid: int, config: Optional[str] = None,
+                  keyring: Optional[str] = None, osd_fsid: Optional[str] = None,
+                  reconfig: Optional[bool] = False, redeploy: Optional[bool] = False,
+                  ports: Optional[List[int]] = None) -> None:
 
     ports = ports or []
-    if any([port_in_use(ctx, port) for port in ports]):
-        if daemon_type == 'mgr':
-            # non-fatal for mgr when we are in mgr_standby_modules=false, but we can't
-            # tell whether that is the case here.
-            logger.warning(
-                f"ceph-mgr TCP port(s) {','.join(map(str, ports))} already in use"
-            )
-        else:
-            raise Error("TCP Port(s) '{}' required for {} already in use".format(','.join(map(str, ports)), daemon_type))
+    # only check port in use if not reconfig or redeploy since service
+    # we are redeploying/reconfiguring will already be using the port
+    if not reconfig and not redeploy:
```

Comment:
I have a slight pet peeve about booleans used this way. Long story short: what does it mean if the caller passes both reconfig=True and redeploy=True ? Right now they basically have this function act in the same way but what about the future. Since these are IIRC mutually exclusive states, the code should try to reflect that (as reasonably) possible. Why not add a enum? Enums are already used in cephadm.  Something like:

```python
class DeploymentKind(Enum):
    DEFAULT = 0
    RECONFIG = 1
    REDEPLOY = 2

def deploy_daemon(..., kind: DeploymentKind = DeploymentKind.DEFAULT, ...)
```

This is food-for-thought, maybe next time not a hard requirement BTW.

Reply:
+1 for this suggestion as these options are in fact mutually exclusive

Reply:
added another commit that converts these bools to an enum class as recommended here